### PR TITLE
Begin Documenting Poketch System

### PR DESCRIPTION
**Pull request by TrainerRiley**
This struct is significantly more complicated than `PoketchData` was; I suspect it's going to be a while before it's fully done. I also renamed the header for the pedometer app because that was what I originally started trying to document, but I think I have to figure out some of these more general Poketch files before I can work on the apps.

### CODE REVIEW

File: src/overlay025/poketch_system.c [hunk 1]
```diff
@@ -0,0 +1,790 @@
...
+    return sub_0203D158();
+}
+
+
+void PoketchSystem_Create(FieldSystem *fieldSys, PoketchSystem **poketchSys, SaveData *saveData, BGL *param3, NNSG2dOamManagerInstance *param4)
+{
+    PoketchSystem *new_system;
+
+    Heap_Create(3, 7, 49152);
+    Heap_Create(3, 8, 49152);
```
Comment (lhearachel) [inline src/overlay025/poketch_system.c:128]: From `include/constants/heap.h`, heap 3 is `HEAP_ID_APPLICATION`.

Are heaps 7 and 8 initialized anywhere else? Can we define constants for them now?

Can we also define a constant for `49152` in `constants/heap.h`?

File: src/overlay025/poketch_system.c [hunk 2]
```diff
@@ -0,0 +1,790 @@
...
+
+static PoketchSystem * ov25_02253CE0 (void)
+{
+    return sub_0203D158();
+}
+
+
+void PoketchSystem_Create(FieldSystem *fieldSys, PoketchSystem **poketchSys, SaveData *saveData, BGL *param3, NNSG2dOamManagerInstance *param4)
+{
+    PoketchSystem *new_system;
```
Comment (lhearachel) [inline src/overlay025/poketch_system.c:125]: Can this be declared inline below?

```c
PoketchSystem *sys = Heap_AllocFromHeap(7, sizeof(PoketchSystem));
```

File: src/overlay025/poketch_system.c [hunk 3]
```diff
@@ -0,0 +1,790 @@
...
+    {POKETCH_APPID_POKEMONHISTORY,      FS_OVERLAY_ID(overlay54)}
+};
+
+static PoketchSystem * ov25_02253CE0 (void)
+{
+    return sub_0203D158();
+}
+
+
+void PoketchSystem_Create(FieldSystem *fieldSys, PoketchSystem **poketchSys, SaveData *saveData, BGL *param3, NNSG2dOamManagerInstance *param4)
```
Comment (lhearachel) [inline src/overlay025/poketch_system.c:123]: Names for `param3` and `param4`.

File: src/overlay025/poketch_system.c [hunk 4]
```diff
@@ -0,0 +1,790 @@
...
+        u32 step_count = PoketchData_StepCount(poketchSys->poketchData);
+
+        if (++step_count > 99999) {
+            step_count = 0;
+        }
+
+        PoketchData_SetStepCount(poketchSys->poketchData, step_count);
+        poketchSys->pedometerUpdated = TRUE;
+    }
+    break;
```
Comment (lhearachel) [inline src/overlay025/poketch_system.c:184]: Is the additional block needed? Can this `u32` be declared outside the switch block to remove the additional scope block?

File: src/overlay025/poketch_system.c [hunk 5]
```diff
@@ -0,0 +1,790 @@
...
+        poketchSys->unk_02 = 0;
+        poketchSys->unk_08 = 0;
+        poketchSys->buttonDir = BUTTON_UP;
+
+        if (ov25_02254284(poketchSys)) {
+            return 1;
+        }
+    }
+
+    return 0;
```
Comment (lhearachel) [inline src/overlay025/poketch_system.c:225]: Use `TRUE` and `FALSE` consts when the function returns `BOOL`.

File: src/overlay025/poketch_system.c [hunk 6]
```diff
@@ -0,0 +1,790 @@
...
+    static void(*const v0[])(PoketchSystem *) = {
+        ov25_02253EA4,
+        ov25_02253F2C,
+        ov25_0225406C,
+        ov25_022540D8
+    };
+
+    PoketchSystem *poketchSys;
+
+    poketchSys = (PoketchSystem *)system;
```
Comment (lhearachel) [inline src/overlay025/poketch_system.c:248]: Condense these lines.

File: src/overlay025/poketch_system.c [hunk 7]
```diff
@@ -0,0 +1,790 @@
...
+}
+
+static void PoketchSystem_MainTask(SysTask *task, void *system)
+{
+    static void(*const v0[])(PoketchSystem *) = {
+        ov25_02253EA4,
+        ov25_02253F2C,
+        ov25_0225406C,
+        ov25_022540D8
+    };
```
Comment (lhearachel) [inline src/overlay025/poketch_system.c:244]: Odd construction. Can we declare a one-time-use function-pointer typedef to make it clearer what this table is meant to represent?

File: src/overlay025/poketch_system.c [hunk 8]
```diff
@@ -0,0 +1,790 @@
...
+static void PoketchSystem_LoadApp(PoketchSystem *poketchSys, int appID)
+{
+    GF_ASSERT(appID >= 0 && appID < NELEMS(sAppOverlayIDs));
+
+    if (poketchSys->loadedAppID == POKETCH_APPID_NONE) {
+
+        for (int i = 0; i < NELEMS(sAppOverlayIDs); i++) {
+
+            if (sAppOverlayIDs[i].appID == appID) {
+
```
Comment (lhearachel) [inline src/overlay025/poketch_system.c:482]: Unnecessary newlines.

File: src/overlay025/poketch_system.c [hunk 9]
```diff
@@ -0,0 +1,790 @@
...
+    poketchSys->unk_02 = 3;
+}
+
+static BOOL ov25_02254228 (PoketchSystem *poketchSys)
+{
+    if (poketchSys->unk_02 == 0) {
+        return 1;
+    }
+
+    return 0;
```
Comment (lhearachel) [inline src/overlay025/poketch_system.c:531]: `TRUE`/`FALSE` consts.

**Review by lhearachel — CHANGES_REQUESTED** [review]

File: include/overlay025/poketch_system.h [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    BUTTON_DOWN,
+};
+
+typedef struct PoketchSystem PoketchSystem;
+
+typedef BOOL (* UnkFuncPtr_ov25_02254238)(void **, PoketchSystem *, BGL *, u32);
+typedef void (* UnkFuncPtr_ov25_02254238_1)(void *);
+typedef void (* UnkFuncPtr_ov25_02254274)(void *);
+
+typedef struct PoketchSystem {
```
[the remark this answers is not in the document]
Comment (TrainerRiley) [reply include/overlay025/poketch_system.h:27]: This doesn't compile. The struct contains a `**PoketchSystem` and one of each of the function pointers, so they must be declared before the struct.

File: src/overlay025/poketch_system.c (hunk 4 above)
Comment (TrainerRiley) [reply src/overlay025/poketch_system.c:182]: Moving the u32 causes the checksum to fail sadly.

**Comment by TrainerRiley** [comment]
Alright, I think it's ready for another look now.

File: src/overlay025/poketch_system.c (hunk 4 above)
Comment (lhearachel) [reply src/overlay025/poketch_system.c:182]: Unfortunate. Could you reorganize the block a bit, then, like so?

```c
    case 5: {
        u32 step_count = PoketchData_StepCount(poketchSys->poketchData);

        if (++step_count > 99999) {
            step_count = 0;
        }

        PoketchData_SetStepCount(poketchSys->poketchData, step_count);
        poketchSys->pedometerUpdated = TRUE;

        break;
    }
```

File: src/overlay025/poketch_system.c [hunk 11]
```diff
@@ -0,0 +1,785 @@
...
+int ov25_02254548 (const PoketchSystem *poketchSys)
+{
+    TrainerInfo * v0 = SaveData_GetTrainerInfo(poketchSys->saveData);
+
+    if (TrainerInfo_Gender(v0) == 1) {
+        return 0;
+    } else {
+        return 1;
+    }
+}
```
Comment (lhearachel) [inline src/overlay025/poketch_system.c:785]: Newline at EOF.

**Review by lhearachel — COMMENTED** [review]
LGTM save for two little nitpicks.

File: src/overlay025/poketch_system.c (hunk 11 above)
Comment (lhearachel) [reply src/overlay025/poketch_system.c:785]: Not me misreading the github diff early in the morning haHAA

**Pull request merged**